mycroft/client/enclosure/enclosure.py

```diff
@@ -202,16 +202,91 @@ class Enclosure:
     """
 
     def __init__(self):
-        self.__init_serial()
+        try:
+            self.config = ConfigurationManager.get().get("enclosure")
+
+            platform = self.config.get('platform')
+            if platform is None:
+                # Use the serial port to check if this is a Mycroft
+                # Mark 1 unit.
+                platform = self.detect_platform()
+                if platform == 'unknown':
+                    # Since this is a semi-permanent detection, be
+                    # certain!  Sometimes noise on the serial line
+                    # causes a faulty mis-detection on a real
+                    # Mycroft Mark 1
+                    platform = self.detect_platform()
+
+                ConfigurationManager.set('enclosure', 'platform',
+                                         platform)
+        except Exception as e:
+            self.disconnect()
+            raise Exception("Exception: Unable to determine platform\n" +
+                            str(e))
+
+        LOGGER.info("Platform = '" + platform + "'")
+        if platform == "mycroft_mark_1":
+            # We are a mycroft_mark_1 unit, start up the
+            # enclosure client to communicate over the
+            # serial port
+            self.__init_serial()
+        else:
+            self.disconnect()
+            raise Exception("Exception: Not a Mycroft Mark 1, shutting down")
+
         self.client = WebsocketClient()
         self.reader = EnclosureReader(self.serial, self.client)
         self.writer = EnclosureWriter(self.serial, self.client)
+
+        # Create helpers to handle the various parts of the enclosure
         self.eyes = EnclosureEyes(self.client, self.writer)
         self.mouth = EnclosureMouth(self.client, self.writer)
         self.system = EnclosureArduino(self.client, self.writer)
+
+        # TODO: Remove EnclosureWeather once the Skill can send images
+        #       directly to the EnclosureAPI.
         self.weather = EnclosureWeather(self.client, self.writer)
         self.__register_events()
 
+    def detect_platform(self):
```

Comment:
We should use the reader and writer to do this sort of negotiation. Also, we can just get enclosure platform as a ping operation. So, python sends "system.platform", then arduino answers it as its current platform == "Mark 1" for instance.
Then, python will read it and confirm. I'll right the flow we should do instead of all this.